stix/idl/processing/imaging/fit_map_2dgauss.pro

```diff
@@ -0,0 +1,53 @@
+;+
+; NAME:
+;    fit_map_2dgauss
+;
+; PURPOSE:
+;    Fits a 2D-Gaussian on a previously computed map
+;
+; CALLING SEQUENCE:
+;    fit_map_2dgauss, in_map [, /do_plot]
+;
+; INPUTS:
+;    in_map : a map structure, as returned by imaging functions
+;
+; OPTIONAL KEYWORD:
+;   do_plot : if set, overplots contours at 25%, 50% and 75% of the Gaussian peak on the map
+;             (note that the map has to be already plotted in the active window) 
+;
+; OUTPUTS:
+;   None. The centroid positions obtained from Gaussian fitting is printed on screen, in both
+;         helioprojective cartesian coordinates and heliographic (Stonyhurst) longitude and latitude
+;
+; MODIFICATION HISTORY:
+;    2022-01-07: F. Schuller (AIP, Germany): created
+;
+;-
+pro fit_map_2dgauss, in_map, do_plot=do_plot
+  ; Dimensions of the map array
+  if not is_struct(in_map) then begin
+    print,"ERROR: input variable is not a structure."
+    return
+  endif
+
+  sz  = size(in_map.data)
+  nbX = sz[1]  &  nbY = sz[2]
+  
+  ; Fit a 2D-Gaussian to the data
+  res = gauss2dfit(in_map.data, coeff, /tilt)
+  ; Here are the center coordinates
+  gauss_X = in_map.xc + in_map.dx*(coeff[4] - nbX/2.)
+  gauss_Y = in_map.yc + in_map.dY*(coeff[5] - nbY/2.)
```

Comment:
The only assumptions is that there is one source that dominates and that can be relatively well fitted by a 2D Gaussian, or do you mean anything else?
Anyway, it's not really part of the imaging pipeline, but since I'm using it a lot, I found it convenient to include it here. I can remove it and keep it in my "private" folder if you wish.

Reply:
I was speaking about the conversion of the pixel to world coordinates it probably fine but `in_map.xc + in_map.dx(pixelx_coord)` isn't always valid